services/py-api/src/utils.py

```diff
@@ -43,3 +45,40 @@ async def stop(self) -> None:
             raise RuntimeError("The AsyncClient has not been initialized!")
 
         await self._async_client.aclose()
+
+
+class JwtUtility:
+    """A class providing methods for encoding data in with a predefined format into a JWT token, or decoding a JWT
+    token into a predefined format (TypedDict)"""
+
+    @staticmethod
+    def encode_data(data: JwtUserData) -> str:
+        return str(encode(data, key="secret", algorithm="HS256"))
+
+    @staticmethod
+    def decode_data(token: str) -> Result[JwtUserData, str]:
```

Comment:
Is this a common pattern within the service? Returning a string instead of an error? I can see how this is more efficient but I think it would be really nice if you establish a proper convention as to what type should always be returned. In most cases, such as in Rust, that would be an Exception. I'm pretty sure I've seen exception being returned instead of strings.

Reply:
yeah, usually we return Err(Exception(...)), but I thought we could use the err messages directly in order to get displayed in the fronted.